### DISCUSSION TITLE
How to re-index assets?

### DISCUSSION THREAD

**Discussion by EtienneT** [dagster-io/dagster#14689] [Q&A] [answered]
Let's say I already have a bunch of files that are each a bit computationally intensive to produce.  They were produced using our legacy software.  We are in the process of porting the code to dagster.  There's one file per day.

I created a new daily partionned dagster asset and ported the legacy code that would produce this asset to the new dagster asset.  My io_manager is saving and loading the files with exactly the same name on disk.  The files are in $DAGSTER_HOME/storage/{asset_name}.  dagit is not detecting the partitions even though.  What could I do to make it see them?  The command `>dagster instance reindex` did nothing.

Thanks!

**Comment by EtienneT** [comment, marked as answer]
I managed to produce this code, but I am not sure if this is the best way.

```python
@op
def re_index_assets(context):
    DAGSTER_HOME = os.environ.get('DAGSTER_HOME')
    p = Path(f'{DAGSTER_HOME}\storage')
    assets = ['price_daily', 'price_weekly']

    for asset in assets:
        for file in (p / asset).glob('**/*.feather'):
            ev = AssetMaterialization(asset_key=asset, partition=file.stem)
            context.log_event(ev)

@job
def re_index_assets_job():
    re_index_assets()
```

**Reply by sryza** [reply]
This is exactly what I would recommend
